### PR TITLE
Adds JSDOC

### PR DESCRIPTION
**Pull request by mlzummo**
@GordonSmith Please Review

### CODE REVIEW

File: src/amchart/Area.js [hunk 1]
```diff
@@ -1,3 +1,8 @@
+/**
```
Comment (GordonSmith) [inline src/amchart/Area.js]: Is this block needed - seems to me we have all this information already?

File: src/amchart/Area.js [hunk 2]
```diff
@@ -6,35 +11,73 @@
         root.amchart_Area = factory(root.d3, root.amchart_CommonSerial, root.amcharts, root.api_INDChart);
     }
 }(this, function(d3, CommonSerial, AmCharts, INDChart) {
+    /**
+     * @class amchart_Area
+     * @extends amchart_CommonSerial
+     * @extends api_INDChart
```
Comment (GordonSmith) [inline src/amchart/Area.js]: It just implements INDChart - it doesn't extend it.

File: src/amchart/Area.js [hunk 3]
```diff
@@ -6,35 +11,73 @@
...
+     * @implements api_INDChart
+     */
     function Area() {
         CommonSerial.call(this);
+        /**
+         * Specifies the class name of the container.
+         * @member {string} _class
+         * @memberof amchart_Area
+         * @private
+         */
```
Comment (GordonSmith) [inline src/amchart/Area.js]: Do we want to document "private" members (anything with an "_")?

File: src/amchart/Area.js [hunk 4]
```diff
@@ -6,35 +11,73 @@
...
         this._gType = "line";
     }
 
     Area.prototype = Object.create(CommonSerial.prototype);
     Area.prototype.implements(INDChart.prototype);
 
-    /**
-     * Publish Params Common To Other Libraries
-     */
+    // Publish Params Common To Other Libraries
```
Comment (GordonSmith) [inline src/amchart/Area.js]: Remove this comment as changes to other files could make it invalid...

File: src/amchart/Area.js [hunk 5]
```diff
@@ -6,35 +11,73 @@
...
+
     Area.prototype.publish("paletteID", "default", "set", "Palette ID", Area.prototype._palette.switch(),{tags:['Basic','Shared']});
     Area.prototype.publish("isStacked", false, "boolean", "Stack Chart",null,{tags:['Basic','Shared']});
     Area.prototype.publish("fillOpacity", 0.7, "number", "Opacity of The Fill Color", null, {min:0,max:1,step:0.001,inputType:'range',tags:['Intermediate','Shared']});
 
+    // Publish Params Unique To This Widget
```
Comment (GordonSmith) [inline src/amchart/Area.js]: Remove this comment as changes to other files could make it invalid...

File: src/amchart/Area.js [hunk 6]
```diff
@@ -6,35 +11,73 @@
...
 
-    /**
-     * Publish Params Unique To This Widget
-     */
     Area.prototype.publish("tooltipTemplate","[[category]]: [[value]]", "string", "Tooltip Text",null,{tags:['Intermediate']});
     Area.prototype.publish("stackType", "regular", "set", "Stack Type",["none","regular","100%"],{tags:['Basic']});
 
+    /**
+     * The function that is executed on first render.
```
Comment (GordonSmith) [inline src/amchart/Area.js]: The function that is called when this widget "enters" the web page.

File: src/amchart/Area.js [hunk 7]
```diff
@@ -6,35 +11,73 @@
...
+     * @method enter
+     * @memberof amchart_Area
+     * @instance
+     * @private
```
Comment (GordonSmith) [inline src/amchart/Area.js]: Should be @protected?

File: src/amchart/Area.js [hunk 8]
```diff
@@ -6,35 +11,73 @@
...
-     * Publish Params Unique To This Widget
-     */
     Area.prototype.publish("tooltipTemplate","[[category]]: [[value]]", "string", "Tooltip Text",null,{tags:['Intermediate']});
     Area.prototype.publish("stackType", "regular", "set", "Stack Type",["none","regular","100%"],{tags:['Basic']});
 
+    /**
+     * The function that is executed on first render.
+     * @method enter
+     * @memberof amchart_Area
+     * @instance
```
Comment (GordonSmith) [inline src/amchart/Area.js]: What does @instance mean?

File: src/amchart/Area.js [hunk 9]
```diff
@@ -6,35 +11,73 @@
...
+     * @private
+     * @param {HTMLElement} domeNode HTML DOMNode of widget container.
```
Comment (GordonSmith) [inline src/amchart/Area.js]: Could also be a SVGElement / SVG DOMNode

File: src/amchart/Area.js [hunk 10]
```diff
@@ -6,35 +11,73 @@
...
+     * @param {D3Selection} element d3 selection object of widget.
+     * @returns {undefined}
```
Comment (GordonSmith) [inline src/amchart/Area.js]: Should be omitted for methods that don't return anything.

File: src/amchart/Area.js [hunk 11]
```diff
@@ -6,35 +11,73 @@
...
     Area.prototype.enter = function(domNode, element) {
         CommonSerial.prototype.enter.apply(this, arguments);
     };
 
+    /**
+     * Updates underlying AmChart widget object, with options from publish parameters.
+     * @method updateChartOptions
+     * @memberof amchart_Area
+     * @instance
+     * @private
```
Comment (GordonSmith) [inline src/amchart/Area.js]: Should be @protected?

File: src/amchart/Area.js [hunk 12]
```diff
@@ -54,6 +97,15 @@
...
     };
 
+    /**
+     * Builds AmChart graph object that becomes a property of the AmChart widget object.
+     * @method buildGraphs
+     * @memberof amchart_Area
+     * @instance
+     * @private
+     * @param {string} gType Value from this._gType.
+     * @returns {Widget}
```
Comment (GordonSmith) [inline src/amchart/Area.js]: Doesn't return anything...

File: src/amchart/Area.js (hunk 1 above)
Comment (mlzummo) [inline src/amchart/Area.js]: yes this is needed if you want the file list to show up with your documentation

File: src/amchart/Area.js (hunk 2 above)
Comment (mlzummo) [inline src/amchart/Area.js]: most templates will not compile the interfaces correctly so this was needed to make sure the interfaces methods and the interface it self shows up.

File: src/amchart/Area.js (hunk 3 above)
Comment (mlzummo) [inline src/amchart/Area.js]: it can come in handy so yes, i'd rather be all inclusive

File: src/amchart/Area.js (hunk 8 above)
Comment (mlzummo) [inline src/amchart/Area.js]: if you want your methods to show up in your class and propagate to its children you have to tell it is an instance method/variable etc...(same goes for your interfaces)

File: src/amchart/Area.js (hunk 2 above)
Comment (GordonSmith) [inline src/amchart/Area.js]: Can we fix the template as its not just a matter of "showing up" it needs to be correct when it does show up!

**Pull request closed without merging**